src/Functions/FunctionsStringArray.h

```diff
@@ -239,21 +244,37 @@ class SplitByCharImpl
     Pos end;
 
     char sep;
+    Int64 max_split = -1;
+    Int64 curr_split = 0;
 
 public:
     static constexpr auto name = "splitByChar";
     static String getName() { return name; }
-    static size_t getNumberOfArguments() { return 2; }
+    static bool isVariadic() { return true; }
+    static size_t getNumberOfArguments() { return 0; }
 
     static void checkArguments(const DataTypes & arguments)
     {
+        if (arguments.size() < 2 || arguments.size() > 3)
+            throw Exception(
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Function '{}' needs at least 2 arguments, at most 3 arguments; passed {}.",
+                arguments.size());
+
         if (!isString(arguments[0]))
             throw Exception("Illegal type " + arguments[0]->getName() + " of first argument of function " + getName() + ". Must be String.",
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
         if (!isString(arguments[1]))
             throw Exception("Illegal type " + arguments[1]->getName() + " of second argument of function " + getName() + ". Must be String.",
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        if (arguments.size() == 3 && !isNativeInteger(arguments[2]))
```

Comment:
Here, we need to check if column is constant, and isNativeInteger.

Reply:
const checking of `argument[2]` is in function `init`, together with `arguments[0]`.